mlflow/pyfunc/__init__.py

```diff
@@ -369,11 +369,11 @@ def _load_model_env(path):
 
 
 def _validate_params(params, model_metadata):
-    if params:
-        if hasattr(model_metadata, "get_params_schema"):
-            params_schema = model_metadata.get_params_schema()
-            return _enforce_params_schema(params, params_schema)
-        else:
+    if hasattr(model_metadata, "get_params_schema"):
+        params_schema = model_metadata.get_params_schema()
+        return _enforce_params_schema(params, params_schema)
+    else:
+        if params:
```

Comment:
```suggestion
    if params:
```

I think we can remove this `else`.